src/ng-option-highlight/lib/ng-option-highlight.directive.ts

```diff
@@ -1,22 +1,23 @@
-import * as searchHelper from './search-helper';
-import { AfterViewInit, Directive, ElementRef, Input, OnChanges, Renderer2 } from '@angular/core';
+import { AfterViewInit, Directive, ElementRef, inject, input, OnChanges, Renderer2 } from '@angular/core';
 
 @Directive({
 	selector: '[ngOptionHighlight]',
 })
 export class NgOptionHighlightDirective implements OnChanges, AfterViewInit {
-	@Input('ngOptionHighlight') term: string;
-
+	term = input<string>(undefined, { alias: 'ngOptionHighlight' });
+	private elementRef = inject(ElementRef);
+	private renderer = inject(Renderer2);
 	private element: HTMLElement;
 	private label: string;
 
-	constructor(
-		private elementRef: ElementRef,
-		private renderer: Renderer2,
-	) {
+	constructor() {
 		this.element = this.elementRef.nativeElement;
 	}
 
+	private get _canHighlight() {
```

Comment:
@pavankjadda  I think it is better to have _canHighlight() computed from "term" input. this is the recommended approach with new reactive system from Angular 

```
_canHighlight = computed(()=> this._isDefined(this.term()) && this._isDefined(this.label));
```